<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        
        <link rel="shortcut icon" href="../../../img/favicon.ico">

	<title>Search Engines - LSCC Documentation</title>

        <link href="../../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../../css/font-awesome-4.0.3.css" rel="stylesheet">
        <link rel="stylesheet" href="../../../css/highlight.css">
        <link href="../../../css/base.css" rel="stylesheet">
        <link href="../../../includes/media/mdx_showable.css" rel="stylesheet">
        <link href="../../../includes/media/mdx_showable_user.css" rel="stylesheet">
        <link href="../../../includes/media/slides.css" rel="stylesheet">
        <link href="../../../includes/media/tute.css" rel="stylesheet">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <!-- Main title -->
            <a class="navbar-brand" href="../../..">LSCC Documentation</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            <!-- Main navigation -->
            <ul class="nav navbar-nav">
            
            
                <li >
                    <a href="../../..">Home</a>
                </li>
            
            
            
                <li class="dropdown active">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Tutorials <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        <li >
                            <a href="../../">Overview</a>
                        </li>
                    
                        <li >
                            <a href="../../ngs_overview/NGS_Overview/">NGS Overview</a>
                        </li>
                    
                        <li >
                            <a href="../../using_git/Using_Git/">Using Git and Github for revision control</a>
                        </li>
                    
                        <li >
                            <a href="../../galaxy_101/galaxy_101/">Introduction to Galaxy</a>
                        </li>
                    
                        <li >
                            <a href="../../galaxy-workflows/galaxy-workflows/">Galaxy Workflows</a>
                        </li>
                    
                        <li >
                            <a href="../../var_detect_advanced/var_detect_advanced_background/">Variant Detection Advanced</a>
                        </li>
                    
                        <li >
                            <a href="../../rna_seq_dge_basic/rna_seq_basic_background/">RNA-seq Basic Background</a>
                        </li>
                    
                        <li >
                            <a href="../../rna_seq_dge_basic/rna_seq_basic_tutorial/">RNA-seq Basic</a>
                        </li>
                    
                        <li >
                            <a href="../../rna_seq_dge_advanced/rna_seq_advanced_tutorial/">RNA-seq Advanced</a>
                        </li>
                    
                        <li >
                            <a href="../../rna_seq_exp_design/rna_seq_experimental_design/">RNA-seq Experimental Design</a>
                        </li>
                    
                        <li >
                            <a href="../../hpc/">Introduction to HPC</a>
                        </li>
                    
                        <li >
                            <a href="../../unix/">Introduction to Unix</a>
                        </li>
                    
                        <li >
                            <a href="../../python_overview/python_overview/">Python Overview</a>
                        </li>
                    
                        <li >
                            <a href="">Microbial de novo Assembly for Illumina Data</a>
                        </li>
                    
                        <li >
                            <a href="../../variant_calling_galaxy_1/variant_calling_galaxy_1/">Introduction to Variant Calling using Galaxy</a>
                        </li>
                    
                        <li class="active">
                            <a href="">Identifying proteins from mass spectrometry data</a>
                        </li>
                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Protocols <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        <li >
                            <a href="../../assembly/assembly-protocol/">De Novo Genome Assembly for Illumina Data</a>
                        </li>
                    
                    </ul>
                </li>
            
            
            </ul>

            <!-- Search, Navigation and Repo links -->
            <ul class="nav navbar-nav navbar-right">
                <li >
                    <a rel="next" href="../background_protein_databases/">
                        <i class="fa fa-arrow-left"></i> Previous
                    </a>
                </li>
                <li >
                    <a rel="prev" href="../background_protein_prophet/">
                        Next <i class="fa fa-arrow-right"></i>
                    </a>
                </li>
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#how-search-engines-work">How Search Engines Work</a></li>
        
    
    </ul>
</div></div>
            <div class="col-md-9" role="main">

<h1 id="how-search-engines-work">How Search Engines Work</h1>
<p>When choosing search engine parameters it can be helpful to understand the basic algorithms that most search engines share. The workflow of a typical search engine is roughly as outlined below.</p>
<ol>
<li>The next spectrum in the dataset is loaded for consideration</li>
<li>The spectrum parent mass and the parent mass tolerance is used to select a small set of matching peptides from the database.  This is a crucial step because the number of peptides that fall within the matching mass window will determine the effective size of the search space.  Database search space size also depends on many other factors including<ul>
<li>The size of the protein database</li>
<li>Variable modifications allowed on peptides</li>
<li>Parent ion mass tolerance</li>
<li>Number of allowed missed enzymatic cleavages</li>
<li>Specificity of the enzyme used for digestion</li>
</ul>
</li>
<li>Each of the matching peptides is scored against the spectrum. The nature of this scoring is where search engines typically differ from each other.</li>
<li>The highest scoring <code>peptide spectrum match</code> (PSM) is recorded along with its score.</li>
<li>Some form of global analysis of (PSM) scores is performed in order to determine a threshold of significance</li>
</ol>
<p>There are many excellent presentations online that explain this in more detail.  Although it's old, I recommend <a href="http://www.slideshare.net/ProteomeSoftware/interpreting-msms-results">this presentation by Brian Searle</a></p></div>
        </div>

        <footer class="col-md-12">
            <hr>
            
            <center>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</center>
        </footer>

        <script src="../../../js/jquery-1.10.2.min.js"></script>
        <script src="../../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../../js/highlight.pack.js"></script>
        <script src="../../../js/base.js"></script>
        <script src="../../../includes/media/delayed-jquery.js"></script>
        <script src="../../../includes/media/mdx_showable.js"></script>
    </body>
</html>