<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        
        <link rel="shortcut icon" href="../../../img/favicon.ico">

	<title>RNA-seq Experimental Design - LSCC Documentation</title>

        <link href="../../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../../css/font-awesome-4.0.3.css" rel="stylesheet">
        <link rel="stylesheet" href="../../../css/highlight.css">
        <link href="../../../css/base.css" rel="stylesheet">
        <link href="../../../includes/media/mdx_showable.css" rel="stylesheet">
        <link href="../../../includes/media/mdx_showable_user.css" rel="stylesheet">
        <link href="../../../includes/media/slides.css" rel="stylesheet">
        <link href="../../../includes/media/tute.css" rel="stylesheet">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <!-- Main title -->
            <a class="navbar-brand" href="../../..">LSCC Documentation</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            <!-- Main navigation -->
            <ul class="nav navbar-nav">
            
            
                <li >
                    <a href="../../..">Home</a>
                </li>
            
            
            
                <li class="dropdown active">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Tutorials <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        <li >
                            <a href="../../">Overview</a>
                        </li>
                    
                        <li >
                            <a href="../../ngs_overview/NGS_Overview/">NGS Overview</a>
                        </li>
                    
                        <li >
                            <a href="../../using_git/Using_Git/">Using Git and Github for revision control</a>
                        </li>
                    
                        <li >
                            <a href="../../gvl_launch/gvl_launch/">Launching a Personal GVL Server</a>
                        </li>
                    
                        <li >
                            <a href="../../galaxy_101/galaxy_101/">Introduction to Galaxy</a>
                        </li>
                    
                        <li >
                            <a href="../../galaxy-workflows/galaxy-workflows/">Galaxy Workflows</a>
                        </li>
                    
                        <li >
                            <a href="../../rna_seq_dge_basic/rna_seq_basic_background/">RNA-seq Basic Background</a>
                        </li>
                    
                        <li >
                            <a href="../../rna_seq_dge_basic/rna_seq_basic_tutorial/">RNA-seq Basic</a>
                        </li>
                    
                        <li >
                            <a href="../../rna_seq_dge_advanced/rna_seq_advanced_tutorial/">RNA-seq Advanced</a>
                        </li>
                    
                        <li class="active">
                            <a href="./">RNA-seq Experimental Design</a>
                        </li>
                    
                        <li >
                            <a href="../../hpc/">Introduction to HPC</a>
                        </li>
                    
                        <li >
                            <a href="../../unix/">Introduction to Unix</a>
                        </li>
                    
                        <li >
                            <a href="../../python_overview/python_overview/">Python Overview</a>
                        </li>
                    
                        <li >
                            <a href="">Microbial de novo Assembly for Illumina Data</a>
                        </li>
                    
                        <li >
                            <a href="../../variant_calling_galaxy_1/variant_calling_galaxy_1/">Introduction to Variant Calling using Galaxy</a>
                        </li>
                    
                        <li >
                            <a href="">Variant Detection Advanced</a>
                        </li>
                    
                        <li >
                            <a href="">Identifying proteins from mass spectrometry data</a>
                        </li>
                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Protocols <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        <li >
                            <a href="../../assembly/assembly-protocol/">De Novo Genome Assembly for Illumina Data</a>
                        </li>
                    
                    </ul>
                </li>
            
            
            </ul>

            <!-- Search, Navigation and Repo links -->
            <ul class="nav navbar-nav navbar-right">
                <li >
                    <a rel="next" href="../../rna_seq_dge_advanced/rna_seq_advanced_tutorial/">
                        <i class="fa fa-arrow-left"></i> Previous
                    </a>
                </li>
                <li >
                    <a rel="prev" href="../../hpc/">
                        Next <i class="fa fa-arrow-right"></i>
                    </a>
                </li>
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#rna-seq-experimental-design">RNA-Seq Experimental Design</a></li>
        
            <li><a href="#what-is-rna-seq">What is RNA-seq?</a></li>
        
            <li><a href="#why-is-a-good-experimental-design-vital">Why is a good experimental design vital?</a></li>
        
            <li><a href="#terminology">Terminology</a></li>
        
            <li><a href="#the-importance-of-replicates-to-estimate-variance">The importance of replicates to estimate variance</a></li>
        
            <li><a href="#how-many-replicates-and-how-many-reads-do-i-need">How many replicates and how many reads do I need?</a></li>
        
            <li><a href="#sequencing-options-to-consider">Sequencing options to consider</a></li>
        
            <li><a href="#summary">Summary</a></li>
        
    
    </ul>
</div></div>
            <div class="col-md-9" role="main">

<h1 id="rna-seq-experimental-design">RNA-Seq Experimental Design</h1>
<h2 id="what-is-rna-seq">What is RNA-seq?</h2>
<p>RNA-seq is a method of measuring gene expression using shotgun sequencing. The
process involves reverse transcribing RNA into cDNA, then sequencing fragments
on a high-throughput platform such as Illumina to obtain a large number of
short reads. For each sample, the reads are then aligned to a genome, and the
number of reads aligned to each gene or feature is recorded.</p>
<p>A typical RNA-seq experiment aims to find differentially expressed genes
between two conditions (e.g. up and down-regulated genes in knock-out mice
compared to wild-type mice). RNA-seq can also be used to discover new
transcripts, splice variants, and fusion genes.</p>
<h2 id="why-is-a-good-experimental-design-vital">Why is a good experimental design vital?</h2>
<p>An RNA-seq experiment produces high dimensional data. This means we get a huge
number of observations for a small number of samples. For example, the
expression of ~20,000 genes could be measured for 6 samples (3 knock-out and 3
wild-type). A frequently used approach to analyse RNA-seq data is to fit each
gene to a linear model where for each of the 20,000 genes, parameters need to
be estimated using a small number of observations. To complicate matters, each
measurement of gene expression is comprised of a mix of biological signal and
unwanted noise. Thus, in order to perform a robust statistical analysis, the
methodology must be carefully designed.</p>
<p>Before you begin any RNA-seq experiment, some questions you should ask
yourself are:</p>
<ul>
<li>Why do you expect to find differentially expressed genes in the particular
   tissue?</li>
<li>What types of genes do you expect to find differentially expressed?</li>
<li>What are the sources of variability from your samples?</li>
<li>Where do you expect most of your variation to come from?</li>
</ul>
<p>A coherent experimental design is the groundwork of a successful experiment.
You should invest time and thought in designing a robust experiment as failing
to think this step through can lead to unusable data and wasted time, money,
and effort.</p>
<p>It is also useful to think about the statistical methods you will use to
analyse the data. If you're planning to bring a data analyst or
bioinformatician onboard for data analysis, you should include him or her in
the experimental design stage.</p>
<h2 id="terminology">Terminology</h2>
<p>Before progressing, it may be useful to define some terms which are commonly
used in RNA-seq.</p>
<table>
  <tr>
    <td>Variability:
    <td>A measure of how much the data is spread around. Variance is
    mathematically defined as the average of the squared difference between
    observations and the expected value. Simply put, a larger variance means
    it is harder to identify differentially expressed genes.
  <tr>
    <td>Feature:
    <td>A defined genomic region. Usually a gene in RNA-seq, but can also
    refer to any region such as an exon or an isoform. In RNA-seq, an estimate
    of abundance is obtained for each feature.
  <tr>
    <td>Biological replicates:
    <td>Samples that have been obtained from biologically separate samples.
    This can mean different individual organisms (e.g. tissue samples from
    different mice), different samplings of the same tumour, or different
    population of cells grown separately from each other but originating from
    the same cell-line. For example, the samples obtained from three different
    knock-out mice could be considered biological replicates in a knock-out
    versus wild-type experiment. A biological replicate combines both technical
    and biological variability as it is also an independent case of all the
    technical steps.
  <tr>
    <td>Technical replicates:
    <td>Samples in which the starting biological sample is the same, but the
    replicates are processed separately. For example, if a biological sample
    is divided and two different library preps are processed and sequenced,
    those two samples would be considered technical replicates.
  <tr>
    <td>Covariate:
    <td>The term 'covariate' is often used interchangeably with 'factor' or
    'variable' in RNA-seq. The term refers to a property of the sample which
    may have some influence on gene expression and should be represented in
    the RNA-seq model. Covariates in RNA-seq are often categorical (e.g.
    treatment condition, sex, batch), but continuous factors are also possible
    (e.g. time points, age). A linear model will contain terms to represent
    the relationships between covariates and each sample. Each possible value
    a factor can take is called a level (e.g. 'male' and 'female' are two
    levels in the factor 'sex'). Factors can either be directly of interest
    to the experiment (e.g. treatment condition) or not of interest (also
    known as nuisance variables) (e.g. sex, batch). The purpose of covariates
    is to explain the variance seen in samples.
  <tr>
    <td>Confounding variable:
    <td>A confounding variable is a nuisance variable that is associated with
    the factor of interest. Possible confounding factors should be controlled
    for so they don't interfere with analysis. For example, if all knock-out
    mice samples were harvested in the morning and all wild-type mice samples
    were harvested in the afternoon, the time of sample collection would be a
    confounding factor as the effects from sample collection time and from the
    knock-out cannot be separated.
  <tr>
    <td>Statistical power:
    <td>The ability to identify differentially expressed genes when there
    really is a difference. This is partly dependent on variance and therefore
    is affected by the number of replicates available and sequencing depth.
</table>

<h2 id="the-importance-of-replicates-to-estimate-variance">The importance of replicates to estimate variance</h2>
<p>When performing a differential gene expression analysis, we look at the
expression values of each gene and try to determine if the expression is
significantly different in the different conditions (e.g. knock-out and
wild-type). The ability to distinguish whether a gene is differentially
expressed is partly determined by the estimates of variability obtained by
using multiple observations in each condition.</p>
<p>Variability is present in two forms: technical variability and biological
variability.</p>
<p>Combined biological and technical variability is measured using biological 
replicates. Biological variability is the main source of variability and is 
due to natural variation in the population and within cells. This includes 
different individuals having different levels of a particular gene and the 
stochastic nature of expression levels in different cells.</p>
<p>Technical variability is measured using technical replicates. Technical
variability is often very small compared to biological variability.  Usually
the question is whether an observed difference is greater than the total 
variability (i.e. significant).  As combined variability is measured by 
biological replicates technical replicates are only important if you need to
know the degree of biological variability or technical variability.  An
example of wanting technical variability would be method development. The
main source of technical variation comes from RNA processing and from 
library prep. Variability from sequencing in different flow cells or different
lanes is usually minimal. Generally, creating technical replicates from multiple
library preps is unnecessary for RNA-seq experiments.</p>
<p>The amount of variance between your biological replicates will affect the
outcome of your analysis. Ideally, you aim to have minimal variability between
samples so you only measure the effect of the condition of interest. Too much
variability between samples can drown out the signal of truly differentially
expressed genes. Controlling for possible confounding factors between
conditions is also important to prevent falsely attributing differential
expression to the condition of interest.</p>
<p>Strategies to minimise variation between samples and to control confounding
variables include:</p>
<ul>
<li>choosing organisms from the same litter,</li>
<li>choosing organisms of the same sex if possible,</li>
<li>using a constant sample collection time,</li>
<li>having the same laboratory technician perform each library prep,</li>
<li>randomising samples to prevent a confounding batch effect if all samples
   can't be processed at one time.</li>
</ul>
<p>If variation between samples can not be removed it should be balanced between
conditions of interest as much as possible, and carefully recorded to allow
its effect to be measured and potentially removed during analysis.</p>
<h2 id="how-many-replicates-and-how-many-reads-do-i-need">How many replicates and how many reads do I need?</h2>
<p>Two very common question asked are:</p>
<ul>
<li>how many biological replicates do I need, and</li>
<li>what sequencing depth is needed for each sample</li>
</ul>
<p>in order to have enough statistical power for my RNA-seq experiment?</p>
<p>These questions cannot be precisely answered without a pilot study. A small
amount of data (minimum of two biological replicates for each condition with
at least 10M reads) can estimate the amount of biological variation, which
determines how many biological replicates are required. Performing a pilot
study is highly recommended to estimate statistical power and identify
possible problems before investing more time and money into the project.</p>
<p><a href="http://euler.bc.edu/marthlab/scotty/scotty.php">Scotty</a> is a web-based tool
that uses data generated from a pilot study to optimize a design for
statistical power. With a limited budget, one must balance sequence coverage
and number of biological replicates. Scotty also has a cost estimate feature
which returns the most powerful design within budget constraints.</p>
<p>As a general rule, the number of biological replicates should never be below 3.
For a basic RNA-seq differential expression experiment, 10M to 20M reads per
sample is usually enough.  If similar data exists it can be helpful to check
the read counts for key genes of interest to estimate the required depth.</p>
<p>Biological variability is usually the largest effect limiting the power of 
RNA-seq analysis.  The most improvement in an experiment will usually be 
achieved by increasing the biological replication to improve estimation of 
the biological variation.</p>
<p>It is often possible to design experiments where the analysis is done 
incrementally such that a pilot study is added to with an additional block of
samples or a pool of libraries is sequenced to additional depth. In these cases
care must be taken to balance the design in a manner that each stage is a
valid experiment in its own right.  This can allow a focused question to be 
answered in the first stage, with an ability to either address issues or 
progress to a second stage with additional questions.</p>
<h2 id="sequencing-options-to-consider">Sequencing options to consider</h2>
<p><strong>How much total RNA is needed:</strong><br />
Many sequencing centres such as
<a href="http://www.agrf.org.au/resources/next-gen-resources/ngs-faqs#How much RNA do you need?">AGRF</a>
recommend at least 250ng of total RNA for RNA sequencing. It is possible to go
as low as 100ng of total RNA, but results are not guaranteed. The quality of
RNA is also important when making libraries. A RNA Integrity Number (RIN) is a
number from 1 (poor) to 10 (good) and can indicate how much degradation there
is in the sample. A poor score can lead to over representation at the 3' end
of the transcript and low yield. Samples with low RIN scores (below 8) are
not recommended for sequencing.  Care should also be taken to ensure RIN is
consistent between conditions to avoid confounding this technical effect with
the biological question.</p>
<p><strong>Choosing an enrichment method:</strong><br />
Ribosomal RNA makes up &gt;95% of total cellular RNA, so a preparation for
RNA-seq must either enrich for mRNA using poly-A enrichment, or deplete rRNA.
Poly-A enrichment is recommended for most standard RNA-seq experiments, but
will not provide information about microRNAs and other non-coding RNA species.
In general, ribo-depleted RNA-seq data will contain more noise, however, the
protocol is recommended if you have poor or variable quality of RNA as the 3’
bias of poly-A enrichment will be more pronounced with increased RNA
degradation. The amount of RNA needed for each method differs. For Poly-A
enrichment a minimum of 100ng is needed while for ribo-depletion, a minimum of
200ng is recommended.</p>
<p><strong>Choosing read type:</strong><br />
For basic differential expression analysis RNA-seq experiments, single-end
sequencing is recommended to obtain gene transcript counts. In more advanced
experiments, paired-ends are useful for determining transcript structure and
discovering splice variants.</p>
<p><strong>Choosing strandedness:</strong>  </p>
<p>With a non-directional (unstranded) protocol, there is no way to identify
whether a read originated from the coding strand or its reverse complement.
Non-directional protocols allow mapping of a read to a genomic location, but
not the direction in which the RNA was transcribed. They are therefore used to
count transcripts for known genes, and are recommended for basic RNA-seq
experiments. Directional protocols (stranded) preserve strand information and
are useful for novel transcript discovery.</p>
<p><strong>Multiplexing:</strong><br />
Multiplexing is an approach to sequence multiple samples in the same
sequencing lane. By sequencing all samples in the same lane, multiplexing can
also minimise bias from lane effects.</p>
<p><strong>Spike-in controls:</strong><br />
RNA-seq spike-in controls are a set of synthetic RNAs of known concentration
which act as negative or positive controls. These controls have been used for
normalisation and quality control, but recent work has shown that the amount
of technical variability in their use dramatically reduces their utility.</p>
<h2 id="summary">Summary</h2>
<ul>
<li>A good experimental design is vital for a successful experiment. If you're
   planning to work with a data analyst or bioinformatician, include them in
   the design stage.</li>
<li>Aim to minimise variability by identifying possible sources of variance in
   your samples.</li>
<li>Biological replicates are important. The number of biological replicates
   you should have should never be below 3. Technical replicates are often
   unnecessary.</li>
<li>Pilot studies are highly recommended for identifying how many replicates
   and how many reads you should have for enough statistical power in your
   experiment.</li>
<li>For basic RNA-seq experiments, poly-A enriched, single-ended, unstranded
   sequencing at depths of 10M to 20M is probably what you want.</li>
</ul></div>
        </div>

        <footer class="col-md-12">
            <hr>
            
            <center>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</center>
        </footer>

        <script src="../../../js/jquery-1.10.2.min.js"></script>
        <script src="../../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../../js/highlight.pack.js"></script>
        <script src="../../../js/base.js"></script>
        <script src="../../../includes/media/delayed-jquery.js"></script>
        <script src="../../../includes/media/mdx_showable.js"></script>
    </body>
</html>