<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        
        <link rel="shortcut icon" href="../../../img/favicon.ico">

	<title>Launching a Personal GVL Server - LSCC Documentation</title>

        <link href="../../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../../css/font-awesome-4.0.3.css" rel="stylesheet">
        <link rel="stylesheet" href="../../../css/highlight.css">
        <link href="../../../css/base.css" rel="stylesheet">
        <link href="../../../includes/media/mdx_showable.css" rel="stylesheet">
        <link href="../../../includes/media/mdx_showable_user.css" rel="stylesheet">
        <link href="../../../includes/media/slides.css" rel="stylesheet">
        <link href="../../../includes/media/tute.css" rel="stylesheet">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <!-- Main title -->
            <a class="navbar-brand" href="../../..">LSCC Documentation</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            <!-- Main navigation -->
            <ul class="nav navbar-nav">
            
            
                <li >
                    <a href="../../..">Home</a>
                </li>
            
            
            
                <li class="dropdown active">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Tutorials <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        <li >
                            <a href="../../">Overview</a>
                        </li>
                    
                        <li >
                            <a href="../../ngs_overview/NGS_Overview/">NGS Overview</a>
                        </li>
                    
                        <li >
                            <a href="../../using_git/Using_Git/">Using Git and Github for revision control</a>
                        </li>
                    
                        <li class="active">
                            <a href="./">Launching a Personal GVL Server</a>
                        </li>
                    
                        <li >
                            <a href="../../galaxy_101/galaxy_101/">Introduction to Galaxy</a>
                        </li>
                    
                        <li >
                            <a href="../../galaxy-workflows/galaxy-workflows/">Galaxy Workflows</a>
                        </li>
                    
                        <li >
                            <a href="../../rna_seq_dge_basic/rna_seq_basic_background/">RNA-seq Basic Background</a>
                        </li>
                    
                        <li >
                            <a href="../../rna_seq_dge_basic/rna_seq_basic_tutorial/">RNA-seq Basic</a>
                        </li>
                    
                        <li >
                            <a href="../../rna_seq_dge_advanced/rna_seq_advanced_tutorial/">RNA-seq Advanced</a>
                        </li>
                    
                        <li >
                            <a href="../../rna_seq_exp_design/rna_seq_experimental_design/">RNA-seq Experimental Design</a>
                        </li>
                    
                        <li >
                            <a href="../../hpc/">Introduction to HPC</a>
                        </li>
                    
                        <li >
                            <a href="../../unix/">Introduction to Unix</a>
                        </li>
                    
                        <li >
                            <a href="../../python_overview/python_overview/">Python Overview</a>
                        </li>
                    
                        <li >
                            <a href="">Microbial de novo Assembly for Illumina Data</a>
                        </li>
                    
                        <li >
                            <a href="../../variant_calling_galaxy_1/variant_calling_galaxy_1/">Introduction to Variant Calling using Galaxy</a>
                        </li>
                    
                        <li >
                            <a href="">Variant Detection Advanced</a>
                        </li>
                    
                        <li >
                            <a href="">Identifying proteins from mass spectrometry data</a>
                        </li>
                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Protocols <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        <li >
                            <a href="../../assembly/assembly-protocol/">De Novo Genome Assembly for Illumina Data</a>
                        </li>
                    
                    </ul>
                </li>
            
            
            </ul>

            <!-- Search, Navigation and Repo links -->
            <ul class="nav navbar-nav navbar-right">
                <li >
                    <a rel="next" href="../../using_git/Using_Git/">
                        <i class="fa fa-arrow-left"></i> Previous
                    </a>
                </li>
                <li >
                    <a rel="prev" href="../../galaxy_101/galaxy_101/">
                        Next <i class="fa fa-arrow-right"></i>
                    </a>
                </li>
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#launching-a-personal-gvl-server-on-the-nectar-research-cloud">Launching a Personal GVL Server on the NeCTAR Research Cloud</a></li>
        
            <li><a href="#tutorial-overview">Tutorial Overview</a></li>
        
            <li><a href="#background">Background</a></li>
        
            <li><a href="#section-1-access-the-nectar-dashboard">Section 1: Access the NeCTAR dashboard</a></li>
        
            <li><a href="#section-2-get-your-cloud-credentials">Section 2: Get your cloud credentials</a></li>
        
            <li><a href="#section-3-launch-your-personal-gvl-instance">Section 3: Launch your personal GVL instance</a></li>
        
            <li><a href="#section-4-access-your-gvl-instance">Section 4: Access your GVL instance</a></li>
        
            <li><a href="#section-5-gvl-services">Section 5: GVL services</a></li>
        
            <li><a href="#section-6-shutting-your-machine-down">Section 6: Shutting your machine down</a></li>
        
    
    </ul>
</div></div>
            <div class="col-md-9" role="main">

<style type="text/css">
    body{
        line-height: 2;
        font-size: 16px;
    }

    ol li{padding: 4px;}
    ul li{padding: 0px;}
    h4 {margin: 30px 0px 15px 0px;}

    div.code {
        font-family: "Courier New";
        border: 1px solid;
        border-color: #999999;
        background-color: #eeeeee;
        padding: 5px 10px;
        margin: 10px;
        border-radius: 5px;
        overflow: auto;
    }

    div.question {
        color: #666666;
        background-color: #e1eaf9;
        padding: 15px 25px;
        margin: 20px;
        font-size: 15px;
        border-radius: 20px;
    }

    div.question h4 {
        font-style: italic;
        margin: 10px 0px !important;
    }
</style>

<p><img src="../media/gvl_logo.jpg" height=100px align=right></p>
<h1 id="launching-a-personal-gvl-server-on-the-nectar-research-cloud">Launching a Personal GVL Server on the NeCTAR Research Cloud</h1>
<hr />
<h2 id="tutorial-overview">Tutorial Overview</h2>
<p>This document guides you to launch your own GVL Analysis platform, with Galaxy,
using your default NeCTAR allocation.</p>
<p>The tutorial will go over:</p>
<ol>
<li>Accessing the NeCTAR dashboard using Australian Access Federation (AAF)
   credentials</li>
<li>Getting your NeCTAR Research Cloud credentials</li>
<li>Launching the GVL image</li>
<li>Accessing your GVL instance</li>
<li>GVL services</li>
<li>Shutting your machine down</li>
</ol>
<hr />
<h2 id="background">Background</h2>
<h4 id="what-is-the-gvl">What is the GVL?</h4>
<p>The Genomics Virtual Laboratory (GVL) is a research platform that can be
deployed on the cloud.</p>
<p><img src="../media/gvl_dashboard.png" height=500px style="display:block; margin-left: auto; margin-right:auto;"></p>
<p>A private GVL server is a virtual machine running on the cloud and contains a
pre-installed suite of tools for performing bioinformatics analyses. It differs
from public GVL servers (such as the
<a href="http://galaxy-tut.genome.edu.au/">Galaxy Tutorial Server</a>,
<a href="http://galaxy-mel.genome.edu.au/">Galaxy Melbourne</a>, and
<a href="http://galaxy-qld.genome.edu.au/">Galaxy Queensland</a>) by providing full
administrative access to the server, as well as the full suite of GVL services,
whereas public GVL servers provide restricted access for security reasons.
For example, public GVL servers do not provide access to the Ubuntu desktop,
the Linux command line or JupyterHub at present.</p>
<p>Accessing the GVL server is completely free on the Australian NeCTAR Research
Cloud, provided that you have a account with NeCTAR with allocated resources.</p>
<h4 id="what-is-nectar">What is NeCTAR?</h4>
<p>The <a href="http://nectar.org.au/">National eResearch Collaboration Tools and Resources project</a>
(NeCTAR) is an Australian programme that provides computing infrastructure
and services to Australian researchers. The NeCTAR Cloud allows us to
deploy virtual machines as a platform for research.</p>
<p>While it is possible to launch the GVL on <a href="https://aws.amazon.com/">Amazon</a>,
you may have to pay Amazon usage charges (the GVL software itself is free).</p>
<hr />
<h2 id="section-1-access-the-nectar-dashboard">Section 1: Access the NeCTAR dashboard</h2>
<ol>
<li>
<p>Login into the NeCTAR dashboard at
    <a href="https://dashboard.rc.nectar.org.au">dashboard.rc.nectar.org.au</a>.<br />
    Only members of the Australian Access Federation (AAF) can access Australian
    Research Cloud resources. Most Australian universities are members of the
    AAF, however, if you belong to an institution that is not a member of AAF,
    <a href="http://genome.edu.au/help/gvl-help">GVL Help</a> may be able to provide you
    with credentials. You also have the option of using a commercial cloud
    provider such as <a href="https://aws.amazon.com/">AWS</a> to host your server.</p>
</li>
<li>
<p>Choose your organisation from the list and login using your credentials.
    If you are from the University of Melbourne, choose 'The University of
    Melbourne' and not 'The University of Melbourne (with ECP)'.</p>
</li>
<li>
<p>Login with your institutional username and password. If this is the first
    time you have accessed the Australian Research Cloud, you must agree to some
    terms and conditions.</p>
<p>When you log in for the first time, you are automatically allocated a trial
project which lasts for 3 months. This trial project allows you to launch a
medium instance (with 2 cores) which is sufficient for launching a GVL
instance.</p>
<p><img src="../media/nectar_dashboard.png" height=400px style="display:block; margin-left: auto; margin-right:auto;"></p>
<p>If you have projects which require more compute resources, you can apply
for more allocation <a href="https://dashboard.rc.nectar.org.au/allocation/">here</a>.</p>
</li>
</ol>
<hr />
<h2 id="section-2-get-your-cloud-credentials">Section 2: Get your cloud credentials</h2>
<p>Launching a GVL instance requires EC2 API keys from NeCTAR. Obtaining these keys
is a simple 3 step process.</p>
<p><img src="../media/get_ec2_annotated.png" height=400px/></p>
<ol>
<li>
<p>From the NeCTAR dashboard, on the left sidebar, navigate to<br />
    Project &gt; Compute &gt; Access &amp; Security</p>
</li>
<li>
<p>Click on the top <strong>'API Access'</strong> tab.</p>
</li>
<li>
<p>Click on the <strong>'+ View Credentials'</strong> button on the top right. A window
    containing your API access key (1) and your secret key (2) will appear.
    Click on the eye icon to view your secret key. Keep this key secret and
    secure.</p>
<p><img src="../media/ec2_creds_annotated.png" height=300px/></p>
<p>Keep this page open for later reference.</p>
</li>
</ol>
<hr />
<h2 id="section-3-launch-your-personal-gvl-instance">Section 3: Launch your personal GVL instance</h2>
<ol>
<li>
<p>In a new browser tab, go to <a href="https://launch.genome.edu.au">launch.genome.edu.au</a></p>
</li>
<li>
<p>Fill in the required fields:</p>
<ul>
<li><strong>Cloud:</strong> Keep the default: NeCTAR (OpenStack)</li>
<li><strong>Access key / Secret key:</strong> Copy and paste your access key and your secret
  key you obtained in the previous step in the required fields.</li>
<li><strong>Institutional email:</strong> Enter your institutional email address</li>
<li><strong>Cluster name:</strong> Choose 'Specify a new name' and enter a name for your
  instance (eg. GVL_workshop). It is recommended you choose a unique name
  if you launch multiple instances.</li>
<li><strong>Password:</strong> Choose a strong password and remember it. This is the
  password you will use later to log into your instance.</li>
<li><strong>Instance type:</strong> Keep the default: Medium (2 vcpu / 8GB RAM)</li>
<li><strong>Cluster type:</strong> Keep the default: Cluster with Galaxy</li>
<li><strong>Storage type:</strong> Keep the default: Transient instance storage</li>
</ul>
<p><img src="../media/launch_gvl.png" height=500px style="display:block; margin-left: auto; margin-right:auto; margin-top:10px;"></p>
</li>
<li>
<p>Optional advanced options<br />
    Toggle the 'Show advanced startup options' option to see more options. For
    this tutorial, it is not necessary to modify any of the advanced options.</p>
<ul>
<li><strong>Placement:</strong> You can also choose the region of where your server is
  hosted. If you are doing lots of data transfer, it may be beneficial to
  pick a location close to your physical location. More information about
  zones can be found <a href="https://support.rc.nectar.org.au/docs/availability-zones">here</a>.</li>
<li><strong>Key pair:</strong> Key pairs are SSH credentials that can be used to access
  your instance. You can create and import key pairs in the NeCTAR dashboard
  by navigating to Project &gt; Compute &gt; Access &amp; Security &gt; Key Pairs and
  creating or importing a key pair.</li>
<li><strong>Image:</strong> The image to start. The default option is usually the latest,
  most stable GVL image.</li>
<li><strong>Flavor:</strong> Flavours are versions of the GVL with slightly customised
  toolsets. These toolsets are optimised for different usage patterns. More
  information about the different flavours can be found
  <a href="http://genome.edu.au/get/get#gvlflavours">here</a>. For this tutorial, keep
  the default option of 'GVL + Tutorial Indices'</li>
</ul>
<p><img src="../media/launch_gvl_advanced_options.png" height=500px style="display:block; margin-left: auto; margin-right:auto; margin-top:10px;"></p>
</li>
<li>
<p>Click <strong>'Create a cluster'</strong> to launch the image.<br />
    The launch process takes 2-5 minutes to start the machine and another 5
    minutes to start and configure Galaxy.</p>
<p><img src="../media/launch_gvl_wait.png" height=250px style="display:block; margin-left: auto; margin-right:auto;"></p>
<p>If your progress bar seems stuck on the 'Requesting' stage for &gt; 5 minutes,
navigate back to the launch page and try selecting a different availability
zone under the advanced startup options <strong>Placement</strong> field. You will need
re-enter your secret key and your password.</p>
</li>
</ol>
<hr />
<h2 id="section-4-access-your-gvl-instance">Section 4: Access your GVL instance</h2>
<ol>
<li>
<p>Once your instance has finished launching, click on the cluster IP address
    to access your GVL dashboard.</p>
<p><img src="../media/launch_gvl_ready.png" height=500px style="display:block; margin-left: auto; margin-right:auto;"></p>
<p>If you accidentally closed the launch page, you can find your cluster's
address on the <a href="https://dashboard.rc.nectar.org.au/project/instances/">NeCTAR dashboard</a>
by navigating to Project &gt; Compute &gt; Instances on the left panel. This page
contains a list of your instances and can be used to terminate your instance
if anything goes wrong. Copy and paste the instance's IP address into your
browser's URL navigation bar.</p>
</li>
<li>
<p>Explore the GVL dashboard.
    Have a read through of the services provided by the GVL.</p>
</li>
</ol>
<p><img src="../media/gvl_dashboard.png" height=500px style="display:block; margin-left: auto; margin-right:auto;"></p>
<hr />
<h2 id="section-5-gvl-services">Section 5: GVL services</h2>
<p>Listed below are short descriptions of the services the GVL provides.</p>
<h4 id="galaxy">Galaxy</h4>
<p><a href="https://galaxyproject.org/">Galaxy</a> is a web-based platform for computational
biomedical research. The GVL has a number of Galaxy tutorials available
<a href="http://genome.edu.au/learn">here</a>.</p>
<p><img src="../media/gvl_galaxy.png" height=400px style="display:block; margin-left: auto; margin-right:auto;"></p>
<p>To begin using Galaxy, register as a user by navigating to User &gt; Register
on the top Galaxy bar.</p>
<h4 id="cloudman">CloudMan</h4>
<p>CloudMan is a cloud manager that manages services on your GVL instance. Use
Cloudman to start and manage your Galaxy service and to add additional nodes
to your compute cluster (if you have enough resources).</p>
<p>You can log into CloudMan by using the username 'ubuntu' and your cluster
password.</p>
<p><img src="../media/cloudman.png" height=350px style="display:block; margin-left: auto; margin-right:auto;"></p>
<p>You can also shut down your instance (permanently) with CloudMan.</p>
<h4 id="lubuntu-desktop">Lubuntu Desktop</h4>
<p>Lubuntu is a lightweight desktop environment through which you can run desktop
applications on your virtual machine through your web browser. You can also
access the GVL command line utilities through the desktop.</p>
<p>You can log into Lubuntu Desktop using the username 'ubuntu' and your cluster
password.</p>
<p><img src="../media/lubuntu.png" height=350px style="display:block; margin-left: auto; margin-right:auto;"></p>
<h4 id="ssh">SSH</h4>
<p>Secure Shell (SSH) is a network protocol that allows us to connect to a remotely
machine. You can login to your virtual machine remotely through an SSH client.</p>
<p>If you are using Windows, you will need to download an SSH client such as
<a href="http://www.chiark.greenend.org.uk/~sgtatham/putty/">PuTTY</a>. If you are using
OSX, open up a Terminal window.</p>
<p>If you are unfamiliar with the command line and UNIX, many
<a href="http://swcarpentry.github.io/shell-novice/">tutorials</a>
<a href="http://vlsci.github.io/lscc_docs/tutorials/unix/">on UNIX</a>
<a href="http://www.ee.surrey.ac.uk/Teaching/Unix/">can be</a>
<a href="http://www.doc.ic.ac.uk/~wjk/UnixIntro/">found online</a>.</p>
<p>You can ssh into your machine using the either the username 'ubuntu' or the
username 'researcher' and using your cluster password. It is recommended to use
the researcher account when you are doing your computational research and use
the ubuntu account when you need administrative powers (such as installing
software).</p>
<p><img src="../media/command_line.png" height=350px style="display:block; margin-left: auto; margin-right:auto;"></p>
<h4 id="jupyterhub">JupyterHub</h4>
<p><a href="http://jupyter.org/">JupyterHub</a> is a web-based interactive computational
environment where you can combine code execution, text, mathematics, plots and
rich media into a single document. Currently, JupyterHub can connect to Python2
and Python3 kernals.</p>
<p>If you are unfamiliar with Python, there are many
<a href="http://swcarpentry.github.io/python-novice-inflammation/">tutorials</a>
<a href="http://vlsci.github.io/lscc_docs/tutorials/python_overview/python_overview/">available</a>
<a href="http://pythonprogramminglanguage.com/">online</a>.</p>
<p>You can log into JupyterHub with the username 'researcher' and your cluster
password.</p>
<p>You may need to install Python packages you intend to use via the command line
beforehand.</p>
<p><img src="../media/jupyterhub.png" height=350px style="display:block; margin-left: auto; margin-right:auto;"></p>
<h4 id="rstudio">RStudio</h4>
<p>RStudio Server gives browser-based access to RStudio, the popular programming
and analysis environment for the R programming language. You can find out more
about RStudio <a href="https://www.rstudio.com/home/">here</a>, and the R programming
language <a href="https://www.r-project.org/">here</a>.</p>
<p>You can log into RStudio with the username 'researcher' and your cluster
password.</p>
<p><img src="../media/rstudio.png" height=350px style="display:block; margin-left: auto; margin-right:auto;"></p>
<h4 id="public-html">Public HTML</h4>
<p>This is a shared web-accessible folder. Any files you place in the directory
<code>/home/researcher/public_html</code> will be publicly accessible.</p>
<hr />
<h2 id="section-6-shutting-your-machine-down">Section 6: Shutting your machine down</h2>
<p>There are two ways to terminate your instance. Terminating your instance is
permanent and all data will be deleted (unless you have persistent
<a href="https://support.rc.nectar.org.au/docs/volumes">volume storage</a> which you will
need to apply for).</p>
<ol>
<li>
<p>Via Cloudman<br />
    Log into CloudMan by using the username 'ubuntu' and your cluster password.
    Click the <strong>Shut down...</strong> button under Cluster Controls.</p>
</li>
<li>
<p>Via the NeCTAR dashboard<br />
    Navigate to the <a href="https://dashboard.rc.nectar.org.au/project/instances/">Instances</a>
    page by navigating to Project &gt; Compute &gt; Instances on the left panel. Find
    the instance you want to terminate, and on the right-most column (Actions),
    click on the arrow button, and select <strong>Terminate Instance</strong>.</p>
</li>
</ol></div>
        </div>

        <footer class="col-md-12">
            <hr>
            
            <center>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</center>
        </footer>

        <script src="../../../js/jquery-1.10.2.min.js"></script>
        <script src="../../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../../js/highlight.pack.js"></script>
        <script src="../../../js/base.js"></script>
        <script src="../../../includes/media/delayed-jquery.js"></script>
        <script src="../../../includes/media/mdx_showable.js"></script>
    </body>
</html>