<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        
        <link rel="shortcut icon" href="../../../img/favicon.ico">

	<title>Var detect advanced background - LSCC Documentation</title>

        <link href="../../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../../css/font-awesome-4.0.3.css" rel="stylesheet">
        <link rel="stylesheet" href="../../../css/highlight.css">
        <link href="../../../css/base.css" rel="stylesheet">
        <link href="../../../includes/media/mdx_showable.css" rel="stylesheet">
        <link href="../../../includes/media/mdx_showable_user.css" rel="stylesheet">
        <link href="../../../includes/media/slides.css" rel="stylesheet">
        <link href="../../../includes/media/tute.css" rel="stylesheet">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>

            <!-- Main title -->
            <a class="navbar-brand" href="../../..">LSCC Documentation</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            <!-- Main navigation -->
            <ul class="nav navbar-nav">
            
            
                <li >
                    <a href="../../..">Home</a>
                </li>
            
            
            
                <li class="dropdown active">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Tutorials <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        <li >
                            <a href="../../">Overview</a>
                        </li>
                    
                        <li >
                            <a href="../../ngs_overview/NGS_Overview/">NGS Overview</a>
                        </li>
                    
                        <li >
                            <a href="../../using_git/Using_Git/">Using Git and Github for revision control</a>
                        </li>
                    
                        <li >
                            <a href="../../gvl_launch/gvl_launch/">Launching a Personal GVL Server</a>
                        </li>
                    
                        <li >
                            <a href="../../galaxy_101/galaxy_101/">Introduction to Galaxy</a>
                        </li>
                    
                        <li >
                            <a href="../../galaxy-workflows/galaxy-workflows/">Galaxy Workflows</a>
                        </li>
                    
                        <li >
                            <a href="../../rna_seq_dge_basic/rna_seq_basic_background/">RNA-seq Basic Background</a>
                        </li>
                    
                        <li >
                            <a href="../../rna_seq_dge_basic/rna_seq_basic_tutorial/">RNA-seq Basic</a>
                        </li>
                    
                        <li >
                            <a href="../../rna_seq_dge_advanced/rna_seq_advanced_tutorial/">RNA-seq Advanced</a>
                        </li>
                    
                        <li >
                            <a href="../../rna_seq_exp_design/rna_seq_experimental_design/">RNA-seq Experimental Design</a>
                        </li>
                    
                        <li >
                            <a href="../../hpc/">Introduction to HPC</a>
                        </li>
                    
                        <li >
                            <a href="../../unix/">Introduction to Unix</a>
                        </li>
                    
                        <li >
                            <a href="../../python_overview/python_overview/">Python Overview</a>
                        </li>
                    
                        <li >
                            <a href="">Microbial de novo Assembly for Illumina Data</a>
                        </li>
                    
                        <li >
                            <a href="../../variant_calling_galaxy_1/variant_calling_galaxy_1/">Introduction to Variant Calling using Galaxy</a>
                        </li>
                    
                        <li class="active">
                            <a href="">Variant Detection Advanced</a>
                        </li>
                    
                        <li >
                            <a href="">Identifying proteins from mass spectrometry data</a>
                        </li>
                    
                    </ul>
                </li>
            
            
            
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Protocols <b class="caret"></b></a>
                    <ul class="dropdown-menu">
                    
                        <li >
                            <a href="../../assembly/assembly-protocol/">De Novo Genome Assembly for Illumina Data</a>
                        </li>
                    
                    </ul>
                </li>
            
            
            </ul>

            <!-- Search, Navigation and Repo links -->
            <ul class="nav navbar-nav navbar-right">
                <li >
                    <a rel="next" href="../var_detect_advanced/">
                        <i class="fa fa-arrow-left"></i> Previous
                    </a>
                </li>
                <li >
                    <a rel="prev" href="../../proteomics_basic/proteomics_basic/">
                        Next <i class="fa fa-arrow-right"></i>
                    </a>
                </li>
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#introduction-to-variant-detection">Introduction to Variant detection</a></li>
        
            <li><a href="#background">Background</a></li>
        
            <li><a href="#variant-calling-software">Variant Calling Software:</a></li>
        
            <li><a href="#the-galaxy-workflow-platform">The Galaxy workflow platform</a></li>
        
            <li><a href="#data-format-used-in-the-tutorial">Data Format used in the tutorial</a></li>
        
    
    </ul>
</div></div>
            <div class="col-md-9" role="main">

<h1 id="introduction-to-variant-detection">Introduction to Variant detection</h1>
<h2 id="background">Background</h2>
<p>A variant is something that is different from a standard or type.</p>
<p>The aim of variation detection is to detect how many bases out of the total are different to a reference genome.</p>
<p>In Craig Venter’s genome 4.1 million DNA variants were reported.</p>
<p>What sort of variation could we find in the DNA sequencing?</p>
<ul>
<li>Single nucleotide variations (SNVs)</li>
<li>Single nucleotide polymorphisms (SNPs)</li>
<li>Small insertions and deletions (INDELs)</li>
<li>Large Chromosome rearrangements-Structural variations (SV)</li>
<li>Copy number variations (CNV)</li>
</ul>
<h3 id="variant-calling-vs-genotyping">Variant Calling vs genotyping</h3>
<p>Variant calling is concerned with whether there is evidence of variant in a particular locus whereas genotyping talks about what the sets of alleles in that locus are and their frequencies. In haploid organisms variant calling and genotyping are equivalent whereas the same rule does not apply to other organisms.</p>
<p>Variant callers estimate the probability of a particular genotype given the observed data.</p>
<p>The question one would be asking is what possible genotypes would be possible for a sample. The remaining question is, given that our variant calling process calls a variant, does that mean that there is truly a variant in this locus and also given that the variant caller doesn’t detect a variant in a position does that mean there is no variant in that position.</p>
<p>The result of variant calling is a list of probable variants.</p>
<h3 id="process-of-variant-calling">Process of variant calling</h3>
<p>Sample DNA -&gt; Sequencing -&gt; Read alignment -&gt; BAM file of aligned reads against reference genome -&gt; Genotyper -&gt; Variant list</p>
<p>The number of reads that stack up on each other is called <em>read coverage</em>. The data is converted into positional information of the reference with the read counts that have piled up under each position. Variant calling will look at how many bases out of the total number of bases is different to the reference at any position.</p>
<h3 id="homozygous-or-heterozygous-mutations">Homozygous or Heterozygous mutations:</h3>
<p>What should be noted about variants is that they are rare events and homozygous variants are even rarer than heterozygous events.</p>
<h2 id="variant-calling-software">Variant Calling Software:</h2>
<p>There a number of software available for variant calling some of which are as follows:</p>
<ol>
<li>SAMtools (mpileup and bcftools): Li 2009 Bioinformatics</li>
<li>GATK: McKenna et al. 2010 Genome Res</li>
<li>FreeBayes: MIT</li>
<li>DiBayes: SOLiD software http://www.lifetechnologies.com</li>
<li>InGAP: Qi J, Zhao F, Buboltz A, Schuster SC.. 2009. <em>Bioinformatics</em></li>
<li>MAQGene: Bigelow H, Doitsidou M, Sarin S, Hobert O. 2009. <em>Nature Methods</em></li>
</ol>
<h3 id="variant-calling-using-samtools-mpileup-bcftools">Variant Calling using Samtools (Mpileup + bcftools)</h3>
<p>Samtools calculates the genotype likelihoods. We then pipe the output to bcftools, which does our SNP calling based on those likelihoods.</p>
<p>Mpileup: Input: BAM file Output: Pileuped up reads under the reference</p>
<p>bcftools: Input: Pileup output from Mpileup Output: VCF file with sites and genotypes</p>
<p><a href="http://samtools.sourceforge.net/mpileup.shtml">Further information</a></p>
<h3 id="variant-calling-using-gatk-unified-genotyper">Variant Calling using GATK-Unified Genotyper</h3>
<p>GATK is a programming framework based on the philosophy of MapReduce for developing NGS tools in a distributed or shared memory parallelized form.</p>
<p>GATK unified genotyper uses a Bayesian probabilistic model to calculate genotype likelihoods.</p>
<p>Inputs: BAM file</p>
<p>Output: VCF file with sites and genotypes.</p>
<p>The probability of a variant genotype for a given sequence of data is calculated using the <a href="http://en.wikipedia.org/wiki/Bayes%27_theorem">Bayes Theorem</a> as follows:</p>
<pre><code>P(Genotype | Data) =  (P(Data | Genotype) * P(Genotype)) / P(Data)
</code></pre>

<p>P(Genotype) is the overall probability of that genotype being present in a sequence. This is called the prior probability of a Genotype.</p>
<p>P(Data | Genotype) is the probability of the data (the reads) given the genotype</p>
<p>P(Data) is the probability of seeing the reads.</p>
<p>GATK unified genotyper is not very good in dealing with INDELs and thus we would only calculate SNPs throughout this tutorial. GATK is setup to work with diploid genomes but can be used on haploids as well.</p>
<p><a href="http://www.broadinstitute.org/gatk/gatkdocs/org_broadinstitute_sting_gatk_walkers_genotyper_UnifiedGenotyper.html">Further information</a></p>
<h3 id="variant-calling-using-freebayes">Variant Calling using FreeBayes</h3>
<p>FreeBayes is a high performance, flexible variant caller which uses the open source Freebayes tool to detect genetic variations based high throughput sequencing data (BAM files).</p>
<p><a href="http://bioinformatics.bc.edu/marthlab/FreeBayes">Further information</a></p>
<h3 id="evaluation-of-detected-variants-using-variant-eval">Evaluation of detected variants using Variant Eval</h3>
<p>The identified variation can further be evaluated against known variations such as common dbSNPs. The result can be checked for high concordance to the common SNPs or a known set of SNPs, the truth set.</p>
<p>The results will have:</p>
<ul>
<li>True Positives (TP): The variants called by the software which are also a known variant in the known variants file.</li>
<li>False Positives (FP): The Variants called by the software which are not known to be variants in the known variants file.</li>
<li>True Negatives (TN): The variants not called by the software which are not known to be variants in the known variants file.</li>
<li>False Negatives (FN): The variants not called by the software which are known as variants in the known variants file.</li>
</ul>
<h3 id="quality-matrix">Quality Matrix:</h3>
<pre><code>TP | FP
---|----
TN | FN
</code></pre>

<p>Sensitivity: TP/(TP+FN)</p>
<p>Specificity: TN/(TN+FP)</p>
<p><em>Note:</em></p>
<p>Although software methods available can find variants in unique regions reliably, the short NGS read length prevent them from detecting variations in repetitive regions with comparable sensitivity.</p>
<p>DNA substitution mutations are of two types: Transitions and Transversions. The Ti/Tv ratio (Transitions/Transversions) is also an indicator of how well the model has performed for genotyping.</p>
<ul>
<li>
<p>Transition: a point mutation in which a purine nucleotide is changed to another purine nucleotide. (A\&lt;->G) or a pyrimidine nucleotide to another pyrimidine. Approximately 2 out of 3 SNPs are Transitions.</p>
</li>
<li>
<p>Transversion: a substitute of a purine for a pyrimidine.</p>
</li>
</ul>
<p>Although there are twice as many Transversions as there are Transitions because of the molecular mechanisms by which they are generated, Transition mutations occur at the higher rate than the Transversion mutations.</p>
<p>For more details on variant eval visit: <a href="http://www.broadinstitute.org/gatk/gatkdocs/org_broadinstitute_sting_gatk_walkers_varianteval_VariantEval.html">http://www.broadinstitute.org/gatk/gatkdocs/org_broadinstitute_sting_gatk_walkers_varianteval_VariantEval.html</a></p>
<p><strong>Notes:</strong></p>
<p>An important thing worth noting is the more data the better the variant calling. In addition multisampling improves performance.</p>
<h3 id="local-realignment">Local realignment</h3>
<p>In order to call SNPs close by INDELs correctly, local realignment is strongly recommended before variant calling when using both UnifiedGenotyper and FreeBayes. Samtools mpileup output would not however be affected since it works around this by introducing Base Alignment Quality (BAC). For more information on BAC refer to: <a href="http://samtools.sourceforge.net/mpileup.shtml">http://samtools.sourceforge.net/mpileup.shtml</a></p>
<h2 id="the-galaxy-workflow-platform">The Galaxy workflow platform</h2>
<p>Galaxy is an online bioinformatics workflow management system. Essentially, you upload your files, create various analysis pipelines and run them, then visualise your results.</p>
<p>Galaxy is really an interface to the various tools that do the data processing; each of these tools could be run from the command line, outside of Galaxy. Galaxy makes it easier to link up the tools together and visualise the entire analysis pipeline.</p>
<p>Galaxy uses the concept of 'histories'. Histories are sets of data and workflows that act on that data. The data for this workshop is available in a shared history, which you can import into your own Galaxy account</p>
<p><a href="http://wiki.g2.bx.psu.edu/">Learn more about Galaxy here</a></p>
<p><strong>The Galaxy interface. Tools on the left, data in the middle, analysis workflow on the right.</strong></p>
<p><img alt="" src="../media/image03.png" /></p>
<h2 id="data-format-used-in-the-tutorial">Data Format used in the tutorial</h2>
<h4 id="sequence-alignment-map-format"><strong>Sequence Alignment Map format</strong></h4>
<p><strong>SAM format</strong></p>
<p>Sequence Alignment/Map format records all information relevant to how a set of reads aligns to a reference genome. A SAM file has an optional set of header lines describing the context of the alignment, then one line per read, with the following format:</p>
<ul>
<li>11 mandatory fields (+ variable number of optional fields)</li>
<li>1 QNAME: Query name of the read</li>
<li>2 FLAG</li>
<li>3 RNAME: Reference sequence name</li>
<li>4 POS: Position of alignment in reference sequence</li>
<li>5 MAPQ: Mapping quality (Phred-scaled)</li>
<li>6 CIGAR: String that describes the specifics of the alignment against the reference</li>
<li>7 MRNM</li>
<li>8 MPOS</li>
<li>9 ISIZE</li>
<li>10 SEQQuery: Sequence on the same strand as the reference</li>
<li>11 QUAL: Query quality (ASCII-33=Phred base quality)</li>
</ul>
<p><strong>SAM example</strong></p>
<pre><code>    SRR017937.312 16 chr20 43108717 37 76M * 0 0
    TGAGCCTCCGGGCTATGTGTGCTCACTGACAGAAGACCTGGTCACCAAAGCCCGGGAAGAGCTGCAGGAAAAGCCG
    ?,@A=A\&lt;5=,@==A:BB@=B9(.;A@B;\&gt;@ABBB@@9BB@:@5\&lt;BBBB9)\&gt;BBB2\&lt;BBB@BBB?;;BABBBBBBB@
</code></pre>

<p>For this example:</p>
<ul>
<li><strong>QNAME = SRR017937.312</strong> - this is the name of this read</li>
<li><strong>FLAG = 16</strong> - see the format description below</li>
<li><strong>RNAME = chr20</strong> - this read aligns to chromosome 20</li>
<li><strong>POS = 43108717</strong> - this read aligns the sequence on chr20 at position 43108717</li>
<li><strong>MAPQ = 37</strong> - this is quite a high quality score for the alignment (b/w 0 and 90)</li>
<li><strong>CIGAR = 76M</strong> - this read aligns to the reference segment across all bases (76 Matches means no deletions or insertions. Note that 'aligns' can mean 'aligns with mismatches' - mismatches that don't affect the alignment are not recorded in this field)</li>
<li><strong>MRNM = *</strong> - see the format description below</li>
<li><strong>MPOS = 0</strong> as there is no mate for this read - the sequenced DNA library was single ended, not mate paired*.</li>
<li><strong>ISIZE = 0</strong> as there is no mate for this read</li>
<li><strong>SEQQuery =</strong> the 76bp sequence of the reference segment</li>
<li><strong>QUAL =</strong> per-base quality scores for each position on the alignment. This is just a copy of what is in the FASTQ file</li>
</ul>
<p><strong>SAM</strong> format is described more fully <a href="http://samtools.sourceforge.net/SAM1.pdf"><em>here</em></a></p>
<p>NOTE: reads are shown mapped to the "sense" strand of the reference, and bases are listed in 5' -> 3' order. This is important because an actual read might be from the other strand of DNA. The alignment tool will try to map the read as it is, and also the reverse compliment. If it was on the other strand then the reverse compliment is shown in the SAM file, rather than the original read itself</p>
<p>See<a href="http://www.illumina.com/technology/paired_end_sequencing_assay.ilmn">http://www.illumina.com/technology/paired_end_sequencing_assay.ilmn</a> for an overview of paired-end sequencing.</p>
<p><strong>SAM file in Galaxy</strong></p>
<p><img alt="" src="../media/image07.png" /></p>
<h4 id="binary-sequence-alignment-map-format">Binary Sequence Alignment Map format</h4>
<p><strong>BAM format</strong></p>
<p>SAM is a text format which is not space efficient. Binary Sequence Alignment is a compressed version of SAM.</p>
<p>Data in a BAM file is binary and therefore can't be visualised as text. If you try and visualise in Galaxy, it will default to downloading the file</p>
<p><strong>BAM file in IGV</strong></p>
<p><img alt="" src="../media/image01.png" /></p>
<h4 id="vcf-file-format">VCF file format</h4>
<p><strong>What is VCF file:</strong></p>
<p>The <a href="http://vcftools.sourceforge.net/specs.html"><em>Variant Call Format</em></a> (VCF) is the emerging standard for storing variant data. Originally designed for SNPs and short INDELs, it also works for structural variations.</p>
<p>VCF consists of a header section and a data section. The header must contain a line starting with one '#', showing the name of each field, and then the sample names starting at the 10th column. The data section is TAB delimited with each line consisting of at least 8 mandatory fields (the first 8 fields in the table below). The FORMAT field and sample information are allowed to be absent. We refer to the official <a href="http://vcftools.sourceforge.net/specs.html"><em>VCF spec</em></a> for a more rigorous description of the format.</p>
<table>
<thead>
<tr>
<th>Col</th>
<th>Field</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>1</td>
<td>CHROM</td>
<td>Chromosome name</td>
</tr>
<tr>
<td>2</td>
<td>POS</td>
<td>1-based position. For an indel, this is the position preceding the indel.</td>
</tr>
<tr>
<td>3</td>
<td>ID</td>
<td>Variant identifier. Usually the dbSNP rsID.</td>
</tr>
<tr>
<td>4</td>
<td>REF</td>
<td>Reference sequence at POS involved in the variant. For a SNP, it is a single base.</td>
</tr>
<tr>
<td>5</td>
<td>ALT</td>
<td>Comma delimited list of alternative seuqence(s).</td>
</tr>
<tr>
<td>6</td>
<td>QUAL</td>
<td>Phred-scaled probability of all samples being homozygous reference.</td>
</tr>
<tr>
<td>7</td>
<td>FILTER</td>
<td>Semicolon delimited list of filters that the variant fails to pass.</td>
</tr>
<tr>
<td>8</td>
<td>INFO</td>
<td>Semicolon delimited list of variant information.</td>
</tr>
<tr>
<td>9</td>
<td>FORMAT</td>
<td>Colon delimited list of the format of individual genotypes in the following fields.</td>
</tr>
<tr>
<td>10+</td>
<td>Sample(s)</td>
<td>Individual genotype information defined by FORMAT.</td>
</tr>
</tbody>
</table>
<h4 id="vcf-format-in-galaxy">VCF format in Galaxy:</h4>
<p><img alt="" src="../media/image06.png" /></p>
<h4 id="bcf-file-format">Bcf file format:</h4>
<h3 id="_1"></h3>
<p><strong>BCF format:</strong></p>
<p>BCF, or the binary variant call format, is the binary version of VCF. It keeps the same information in VCF, while much more efficient to process especially for many samples. The relationship between BCF and VCF is similar to that between BAM and SAM.</p></div>
        </div>

        <footer class="col-md-12">
            <hr>
            
            <center>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</center>
        </footer>

        <script src="../../../js/jquery-1.10.2.min.js"></script>
        <script src="../../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../../js/highlight.pack.js"></script>
        <script src="../../../js/base.js"></script>
        <script src="../../../includes/media/delayed-jquery.js"></script>
        <script src="../../../includes/media/mdx_showable.js"></script>
    </body>
</html>